lms/static/js/dashboard/legacy.js

```diff
@@ -95,29 +124,71 @@
 
          $('.action-unenroll').click(function(event) {
-             var element = $(event.target);
-             var track_info = element.data('track-info');
-             var course_number = element.data('course-number');
-             var course_name = element.data('course-name');
-             var cert_name_long = element.data('cert-name-long');
-             $('#track-info').html(interpolate(track_info, {
-                 course_number: "<span id='unenroll_course_number'>" + course_number + '</span>',
-                 course_name: "<span id='unenroll_course_name'>" + course_name + '</span>',
-                 cert_name_long: "<span id='unenroll_cert_name'>" + cert_name_long + '</span>'
-             }, true));
-             $('#refund-info').html(element.data('refund-info'));
-             $('#unenroll_course_id').val(element.data('course-id'));
-             edx.dashboard.dropdown.toggleCourseActionsDropdownMenu(event);
+             var isPaidCourse = $(event.target).data('course-is-paid-course') === 'True';
+             var certNameLong = $(event.target).data('course-cert-name-long');
+             var enrollmentMode = $(event.target).data('course-enrollment-mode');
+
+             var courseNumber = $(event.target).data('course-number');
+             var courseName = $(event.target).data('course-name');
+             var courseRefundUrl = $(event.target).data('course-refund-url');
+             var showRefundOption = '';
+             var dialogMessageAttr;
+
+             var request = $.ajax({
+                 url: courseRefundUrl,
+                 method: 'GET',
+                 dataType: 'json'
+             });
+             request.success(function(data) {
+                 if (data.success) {
+                     showRefundOption = data.course_refundable_status;
+
+                     dialogMessageAttr = setDialogAttributes(isPaidCourse, certNameLong,
+                                    courseNumber, courseName, enrollmentMode, showRefundOption);
+
+                     $('#track-info').empty();
+                     $('#refund-info').empty();
+
+                     $('#track-info').html(interpolate(dialogMessageAttr['data-track-info'], {
+                         course_number: "<span id='unenroll_course_number'>" + courseNumber + '</span>',
+                         course_name: "<span id='unenroll_course_name'>" + courseName + '</span>',
+                         cert_name_long: "<span id='unenroll_cert_name'>" + certNameLong + '</span>'
+                     }, true));
+
+                     if ('data-refund-info' in dialogMessageAttr) {
+                         $('#refund-info').html(dialogMessageAttr['data-refund-info']);
+                     }
+
+                     $('#unenroll_course_id').val($(event.target).data('course-id'));
+                 } else {
+                     $('#unenroll_error').html(
+                        gettext('Unable to get refund status for course, Try again later')
```

Comment:
Maybe a better wording is `Unable to determine whether we should give you a refund because of System Error. Please try again later`?